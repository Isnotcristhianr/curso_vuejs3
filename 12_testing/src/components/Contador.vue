<script>
export default {
    data: () => ({
        msg: 'Contador',
        count: 0
    }),
    methods: {
        increment() {
            this.count++
        },
        decrement() {
            this.count--
        },
        reset() {
            this.count = 0
        }
    }
}
</script>

<template>
    <div>
        <h2>{{ msg }}: {{ count }}</h2>
        <button id="increment" @click="increment">Incrementar</button>
        <button id="decrement" @click="decrement">Decrementar</button>
        <button id="reset" @click="reset">Reiniciar</button>
    </div>
</template>
