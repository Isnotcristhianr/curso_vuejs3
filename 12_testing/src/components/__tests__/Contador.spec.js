import { mount } from '@vue/test-utils'
import Contador from '../Contador.vue'
import { describe, it } from 'vitest'
import { expect } from 'vitest'

describe('Contador Pruebas', () => {
    it('Debe renderizar el componente correctamente', () => {
        const wrapper = mount(Contador)
        expect(wrapper.html()).toMatchSnapshot()
    })
    
    it('Debe incrementar el contador', async () => {
        const wrapper = mount(Contador)
        await wrapper.find('button#increment').trigger('click')
        expect(wrapper.find('h2').text()).toBe('Contador: 1')
    })

    it('Debe decrementar el contador', async () => {
        const wrapper = mount(Contador)
        await wrapper.find('button#decrement').trigger('click')
        expect(wrapper.find('h2').text()).toBe('Contador: -1')
    })

    it('Debe reiniciar el contador', async () => {
        const wrapper = mount(Contador)
        await wrapper.find('button#reset').trigger('click')
        await wrapper.find('button#increment').trigger('click')
        await wrapper.find('button#increment').trigger('click')
        await wrapper.find('button#increment').trigger('click')
        await wrapper.find('button#increment').trigger('click')
        expect(wrapper.find('h2').text()).toBe('Contador: 4')
    })
    
    
})