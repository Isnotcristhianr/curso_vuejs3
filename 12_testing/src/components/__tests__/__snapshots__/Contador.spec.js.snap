// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Contador Pruebas > Debe renderizar el componente correctamente 1`] = `
"<div>
  <h2>Contador: 0</h2><button id="increment">Incrementar</button><button id="decrement">Decrementar</button><button id="reset">Reiniciar</button>
</div>"
`;
